src/live_restore/live_restore_fs.c

```diff
@@ -0,0 +1,1248 @@
+            __live_restore_fh_close((WT_FILE_HANDLE *)lr_fh, wt_session);
+    }
+    return (ret);
+}
+
+/*
+ * __live_restore_fs_remove --
+ *     Remove a file. We can only delete from the destination directory anyway.
+ */
+static int
+__live_restore_fs_remove(
+  WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *name, uint32_t flags)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_FS *lr_fs;
+    WT_LIVE_RESTORE_FS_LAYER_TYPE layer;
+    WT_SESSION_IMPL *session;
+    char *path;
+    bool exist;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    lr_fs = (WT_LIVE_RESTORE_FS *)fs;
+
+    WT_UNUSED(layer);
+    exist = false;
+    path = NULL;
+
+    WT_RET_NOTFOUND_OK(__live_restore_fs_find_layer(fs, session, name, &layer, &exist));
+    if (ret == WT_NOTFOUND || !exist)
+        return (0);
+
+    /* It's possible to call remove on a file that hasn't yet been created in the destination. In
+     * these cases we only need to create the tombstone */
+    if (layer == WT_LIVE_RESTORE_FS_LAYER_DESTINATION) {
+        WT_ERR(__live_restore_fs_filename(&lr_fs->destination, session, name, &path));
+        lr_fs->os_file_system->fs_remove(lr_fs->os_file_system, wt_session, path, flags);
+    }
+
+    /* We need file tombstones here but can we be sure this is correct? */
+    __live_restore_fs_create_tombstone(fs, session, name, flags);
+    /* We don't have a file handle here so WT must have previously closed it. */
+err:
+
+    __wt_free(session, path);
+    return (ret);
+}
+
+/*
+ * __live_restore_fs_rename --
+ *     Rename a file.
+ */
+static int
+__live_restore_fs_rename(
+  WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *from, const char *to, uint32_t flags)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_FS *lr_fs;
+    WT_LIVE_RESTORE_FS_LAYER_TYPE which;
+    WT_SESSION_IMPL *session;
+    char *path_from, *path_to;
+    bool exist;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    lr_fs = (WT_LIVE_RESTORE_FS *)fs;
+
+    exist = false;
+    path_from = NULL;
+    path_to = NULL;
+
+    /*
+     * WiredTiger frequently renames the turtle file, and some other files. This function is more
+     * critical than it may seem at first.
+     */
+
+    __wt_verbose_debug1(
+      session, WT_VERB_FILEOPS, "LIVE_RESTORE: Renaming file from: %s to %s\n", from, to);
+    WT_RET_NOTFOUND_OK(__live_restore_fs_find_layer(fs, session, from, &which, &exist));
+    if (ret == WT_NOTFOUND || !exist)
+        return (ENOENT);
+
+    /* If the file is the top layer, rename it and leave a tombstone behind. */
+    if (which == WT_LIVE_RESTORE_FS_LAYER_DESTINATION) {
+        WT_ERR(__live_restore_fs_filename(&lr_fs->destination, session, from, &path_from));
+        WT_ERR(__live_restore_fs_filename(&lr_fs->destination, session, to, &path_to));
+        WT_ERR(lr_fs->os_file_system->fs_rename(
+          lr_fs->os_file_system, wt_session, path_from, path_to, flags));
+        __wt_free(session, path_from);
+        __wt_free(session, path_to);
```

Comment:
```suggestion
```
Redundant. We clean them up in after the `err:` label

Reply:
[3f680da](https://github.com/wiredtiger/wiredtiger/pull/11272/commits/3f680da7eafab482a6f15265f1015c817edfa1d5)